src/consensus.rs

```diff
@@ -498,6 +498,34 @@ impl Consensus {
             return Ok(None);
         }
 
+        match self.try_add_origin() {
+            // `try_add_origin` is not applicable:
+            // - either current peer is not an origin peer
+            // - or cluster is already established
+            Ok(false) => (),
+
+            // Successfully proposed origin peer to consensus, return to consensus loop to handle `on_ready`
+            Ok(true) => return Ok(Some(1)),
+
+            // Origin peer is not a leader yet, wait for the next tick and return to consensus loop
+            // to tick Raft node
+            Err(err @ TryAddOriginError::NotLeader) => {
+                log::debug!("{err}");
+
+                let next_tick = previous_tick + tick_period;
+                let duration_until_next_tick = next_tick.saturating_duration_since(Instant::now());
+                thread::sleep(duration_until_next_tick);
```

Comment:
Is this still sound along with <https://github.com/qdrant/qdrant/pull/5173>?

Reply:
I think it is? Should be a no-op, but I'd keep it just in case, so committing origin would be "self-sufficient", if we revert #5173 for some reason.